### PR TITLE
Feat: ping ledger on interval

### PR DESCRIPTION
**Pull request by sharafian**
pings the ledger every 30 seconds to keep the websocket connection alive

### CODE REVIEW

**Review by emschwartz — CHANGES_REQUESTED** [review]
please add a test

File: src/lib/plugin.js [hunk 1]
```diff
@@ -934,4 +937,34 @@ function * requestRetry (requestOptions, retryOptions) {
...
+    clearInterval(pingInterval)
+    ws.removeListener('pong', heartbeat)
+  }
+
+  function heartbeat () {
+    isAlive = true
+  }
+
+  function ping () {
+    if (!isAlive) clear()
```
Comment (sentientwaffle) [inline src/lib/plugin.js]: Should this return? `ping()`-ing again won't do much if the `'pong'` event listener was just removed.

**Review by emschwartz — APPROVED** [review]

File: src/lib/plugin.js [hunk 2]
```diff
@@ -934,4 +937,37 @@ function * requestRetry (requestOptions, retryOptions) {
...
+    ws.removeListener('pong', heartbeat)
+  }
+
+  function heartbeat () {
+    isAlive = true
+  }
+
+  function ping () {
+    if (!isAlive) {
+      clear()
```
Comment (sentientwaffle) [inline src/lib/plugin.js]: Is no `pong` being returned worth logging?

Comment (sentientwaffle) [reply src/lib/plugin.js]: (otherwise this would just silently stop ping-ing)

Comment (sharafian) [reply src/lib/plugin.js]: added a debug message

**Review by sentientwaffle — APPROVED** [review]

**Pull request merged**